packages/neovim/src/attach/attach.test.ts

```diff
@@ -155,12 +155,15 @@ describe('Nvim API', () => {
   });
 
   it('emits "disconnect" after quit', done => {
-    const disconnectMock = jest.fn();
-    nvim.on('disconnect', disconnectMock);
+    let disconnectCalled = false;
+    nvim.on('disconnect', () => {
+      disconnectCalled = true;
+    });
+
     nvim.quit();
 
     proc.on('close', () => {
-      expect(disconnectMock.mock.calls.length).toBe(1);
+      expect(disconnectCalled).toBe(true);
       done();
```

Comment:
since the test is timing out, that means this `done` is not called, which means the 'close' event isn't emitted (or stuff is getting disposed before handling the emitted 'close' event).

Reply:
Yeah, I understand that, but how race could happen to cause this problem?

Reply:
Do we still need the disconnect call on proc ?

Reply:
oh, that causes the test harness (jest) to kill Nvim too early, which then raises EPIPE because Nvim still has messages to process.